src/Mvc/Mvc.NewtonsoftJson/src/NewtonsoftJsonMetadataProvider.cs

```diff
@@ -0,0 +1,58 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Linq;
+using Microsoft.AspNetCore.Mvc.ModelBinding.Metadata;
+using Newtonsoft.Json;
+
+namespace Microsoft.AspNetCore.Mvc.NewtonsoftJson;
+
+/// <summary>
+/// An implementation of <see cref="IDisplayMetadataProvider"/> and <see cref="IValidationMetadataProvider"/> for
+/// the Newtonsoft.Json attribute classes.
+/// </summary>
+internal class NewtonsoftJsonMetadataProvider : IDisplayMetadataProvider, IValidationMetadataProvider
+{
+    /// <summary>
+    /// Initializes a new instance of <see cref="NewtonsoftJsonMetadataProvider"/>.
+    /// </summary>
+    public NewtonsoftJsonMetadataProvider()
+    { }
+
+    /// <inheritdoc />
+    public void CreateDisplayMetadata(DisplayMetadataProviderContext context)
+    {
+        if (context == null)
+        {
+            throw new ArgumentNullException(nameof(context));
+        }
+
+        var propertyName = ReadPropertyNameFrom(context.Attributes);
+
+        if (!string.IsNullOrEmpty(propertyName))
+        {
+            context.DisplayMetadata.DisplayName = () => propertyName;
+        }
+    }
+
+    /// <inheritdoc />
+    public void CreateValidationMetadata(ValidationMetadataProviderContext context)
+    {
+        if (context == null)
+        {
+            throw new ArgumentNullException(nameof(context));
+        }
+
+        var propertyName = ReadPropertyNameFrom(context.Attributes);
+
+        if (string.IsNullOrEmpty(propertyName))
+        {
+            propertyName = context.Key.Name;
```

Comment:
Does this need to go thru NamingStrategy APIs the way it does for STJ to get the camel-cased value?